tower/src/balance/p2c/make.rs

```diff
@@ -12,15 +12,22 @@ use std::{
 };
 use tower_service::Service;
 
-/// Makes `Balancer`s given an inner service that makes `Discover`s.
+/// Construct load balancers over dynamic service sets produced by a wrapped "inner" service.
+///
+/// This is effectively an implementation of [`MakeService`](tower::make_service::MakeService),
+/// except that it forwards the service descriptors (`Target`) to an inner service (`S`), and
+/// expects that service to produce a service set in the form of a [`Discover`]. It then wraps the
+/// service set in a [`Balance`] before returning it as the "made" service.
```

Comment:
The links to `Discover` and `Balance` don't have references...do they actually work? I didn't think RustDoc would link type names in backticks without a reference.

Reply:
Yup: https://github.com/rust-lang/rust/issues/43466. It's on by default on nightly, which is what docsrs uses.